simulator/infra/macro.h

```diff
@@ -177,6 +177,13 @@ static constexpr T ones_ls(const T& value, size_t shamt)
     return ~(~value << shamt);
 }
 
+template<typename T>
+static constexpr T rotate_l(const T& value, size_t shamt)
+{
+    size_t xlen = sizeof(T) * 8;
```

Comment:
Wrooong. For 128 bit numbers, we use Boost types which have `sizeof` greater than 16.
Please use our `bitwidth` macro instead.